src/pages/index.tsx

```diff
@@ -0,0 +1,127 @@
+import Bus from '@/interfaces/DummyData';
+import FormData from '@/interfaces/FormData';
+import { Inter } from 'next/font/google'
+import { useEffect, useState } from 'react'
+
+const inter = Inter({ subsets: ['latin'] })
+
+export default function Home() {
+const [busData, setBusData] = useState<Array<Bus>>([]);
+const [selectedBus, setSelectedBus] = useState<Bus>({} as Bus);
+const [formData, setFormData] = useState<FormData>({
+  bus_number:"",
+  bus_route: "",
+  bus_station: "",
+  bus_people: 0,
+  created_at:new Date(),
+  created_by:1
+});
+
+  useEffect(() => {
+    async function fetchBusData() {
+      const res = await fetch(`api/bus-data`)
+      let busData: Array<Bus> = await res.json();
+      setBusData(busData);
+    }
+    fetchBusData();
+  }, []);
+
+  function onBusChange(data: any) {
+    const bus = JSON.parse(data.target.value)
+    setSelectedBus(bus);
+    formData.bus_number = bus.bus_number;
+    formData.bus_route = "";
+    formData.bus_station = "";
+    formData.bus_people = 0;
+    setFormData({...formData});
+    console.log(formData);
+  } 
+
+  function onRouteChange(data: any) {
+    formData.bus_route = data.target.value;
+    setFormData({...formData});
+  }
+
+  function onStationChange(data: any) {
+    formData.bus_station = data.target.value;
+    setFormData({...formData});
+  }
+
+  function getStation(): Array<string> {
+    if (selectedBus.station_from[0] === formData.bus_route) {
+      return selectedBus.station_to;
+    } 
+    return selectedBus.station_from;
+  }
+
+  function onNrPeopleChange(data: any) {
+    formData.bus_people = data.target.valueAsNumber;
+    setFormData({...formData});
+    console.log(formData);
+  }
+
+  async function saveData(e:any) {
+    e.preventDefault();
+    formData.created_at = new Date();
+    formData.created_by = 1;
+    const res = await fetch(`api/form-data`, {
+      method: "POST",
+    body: JSON.stringify(formData)
+    })
+  }
+
+  return (
+    <>
+    <h1 className="text-center mb-5">Welcome to Bus Sensus</h1>
+    <form onSubmit={saveData}>
+      <div className="mb-4 mx-1">
+        <label htmlFor="busNumber" className="form-label">Bus number</label>
+        <select className="form-select" aria-label="Bus number" id="busNumber" onChange={(event) => onBusChange(event)}>
+         <option value={"{}"}>Please select your bus</option>
+         {busData && busData.map((bus) => {
+          return (
+            <option value={JSON.stringify(bus)} key={bus.id}>
```

Comment:
- in loc de stringify, mai bine pui `bus.id`, sau `bus.bus_number` - in functie de ce ai nevoie sa trimiti la backend

Reply:
- Done